### PR TITLE
Resolve 12

### PR DESCRIPTION
**Pull request by khoinguyen26**
- staff create class

### CODE REVIEW

File: src/main/java/io/github/etases/edublock/rs/model/input/ClassCreate.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        String name,
+        int grade
+) {
+    public boolean validate() {
+        return name != null && !name.isBlank()
+                && grade > 0 && grade < 13;
+    }
+
+    @JsonIgnore
+    public String getClassName() {
```
Comment (HSGamer) [inline src/main/java/io/github/etases/edublock/rs/model/input/ClassCreate.java]: Not needed. `name()` exists.

**Review by HSGamer — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/etases/edublock/rs/model/input/ClassCreate.java (hunk 1 above)
Comment (khoinguyen26) [reply src/main/java/io/github/etases/edublock/rs/model/input/ClassCreate.java]: fixed

File: src/main/java/io/github/etases/edublock/rs/handler/StaffHandler.java [hunk 2]
```diff
@@ -151,4 +151,45 @@ public void handle(Context ctx) {
...
+        }
+
+        @Override
+        public void handle(Context ctx) {
+            ClassCreate input = ctx.bodyValidator(ClassCreate.class)
+                    .check(ClassCreate::validate, "Invalid data")
+                    .get();
+            try (var session = sessionFactory.openSession()) {
+                ResponseWithData<ClassCreate> response = new ResponseWithData<>();
+                if (session.get(Classroom.class, input.name()) != null) {
```
Comment (HSGamer) [inline src/main/java/io/github/etases/edublock/rs/handler/StaffHandler.java]: Use query
`name` is not primary key

**Review by HSGamer — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/etases/edublock/rs/handler/StaffHandler.java [hunk 3]
```diff
@@ -151,4 +151,45 @@ public void handle(Context ctx) {
...
+                int grade = input.grade();
+                var classroom = new Classroom();
+                classroom.setName(name);
+                classroom.setGrade(grade);
+                session.save(classroom);
+                if (transaction.isActive()) {
+                    transaction.commit();
+                } else {
+                    transaction.rollback();
+                }
```
Comment (HSGamer) [inline src/main/java/io/github/etases/edublock/rs/handler/StaffHandler.java]: ??

Where is response?

**Review by HSGamer — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/etases/edublock/rs/handler/StaffHandler.java [hunk 4]
```diff
@@ -151,4 +151,45 @@ public void handle(Context ctx) {
...
+            ClassCreate input = ctx.bodyValidator(ClassCreate.class)
+                    .check(ClassCreate::validate, "Invalid data")
+                    .get();
+            try (var session = sessionFactory.openSession()) {
+                ResponseWithData<ClassCreate> response = new ResponseWithData<>();
+                if (session.get(Classroom.class, input.name()) != null) {
+                    response.setStatus(400);
+                    response.setMessage("Class already exists");
+                    response.setData(input);
+                    ctx.json(response);
```
Comment (HSGamer) [inline src/main/java/io/github/etases/edublock/rs/handler/StaffHandler.java]:
```
ctx.status(400);
response.setStatus(1);
ctx.json(response);
```

**Review by HSGamer — CHANGES_REQUESTED** [review]

**Pull request merged**